### PR TITLE
Fatal warnings API fixes

### PR DESCRIPTION
**Pull request by redorav**
**What does this PR do?**

Splits the fatalwarnings api into fatalwarnings for compiler and linkerfatalwarnings for linker. Changes the special Compile and Link values to All in each api. Implements fatal warnings for the linker in VS2022

**How does this PR change Premake's behavior?**

Changes the fatalwarnings api

**Did you check all the boxes?**

- [x] Focus on a single fix or feature; remove any unrelated formatting or code changes
- [x] Add unit tests showing fix or feature works; all tests pass
- [x] Mention any [related issues](https://github.com/premake/premake-core/issues) (put `closes #XXXX` in comment to auto-close issue when PR is merged)
- [x] Follow our [coding conventions](https://github.com/premake/premake-core/blob/master/CONTRIBUTING.md#coding-conventions)
- [x] Minimize the number of commits
- [x] Align [documentation](https://github.com/premake/premake-core/tree/master/website) to your changes

*You can now [support Premake on our OpenCollective](https://opencollective.com/premake). Your contributions help us spend more time responding to requests like these!*

### CODE REVIEW

File: modules/vstudio/vs2010_vcxproj.lua [hunk 1]
```diff
@@ -2162,9 +2162,39 @@
...
+				-- Create a comma-separated set of warnings to elevate as errors
+				local fatalWarningString = '/wx:'
+			
+				for i, fatal in ipairs(filteredFatalWarnings) do
+					if i ~= 1 then
+						fatalWarningString = fatalWarningString..","
+					end
+
+					fatalWarningString = fatalWarningString..fatal
+				end
```
Comment (Jarod42) [inline modules/vstudio/vs2010_vcxproj.lua:2189]: We have [`table.implode`](https://premake.github.io/docs/table/table.implode/)

File: tests/test_premake.lua [hunk 2]
```diff
@@ -42,31 +42,31 @@
...
 	end
 
 	function suite.hasFatalCompileWarnings()
-		local warnings = { "Compile", "4996" }
+		local warnings = { "All", "4996" }
 		local hasFatal = p.hasFatalCompileWarnings(warnings)
 		test.istrue(hasFatal)
 	end
 
 	function suite.hasFatalCompileWarningsNotPresent()
```
Comment (Jarod42) [inline tests/test_premake.lua:56]: Wonder if we can remove those tests (checking mixing compiler and linker warnings) now.

File: website/docs/linkerfatalwarnings.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+| Value   | Description                   |
+-------------------------------------------
+| All | Treat all linker warnings as errors   |
+
+### Applies To ###
+
+Project configurations.
+
+### Availability ###
```
Comment (Jarod42) [inline website/docs/linkerfatalwarnings.md:23]: Adding the example section with filter usage would be fine IMO.

File: modules/vstudio/vs2010_vcxproj.lua (hunk 1 above)
Comment (redorav) [reply modules/vstudio/vs2010_vcxproj.lua:2189]: Nice, I'll use that instead

File: website/docs/linkerfatalwarnings.md (hunk 3 above)
Comment (redorav) [reply website/docs/linkerfatalwarnings.md:23]: Ok, not a bad idea, I'll come up with a meanigful one (i.e. one that has a translation into other toolsets)

File: tests/test_premake.lua (hunk 2 above)
Comment (redorav) [reply tests/test_premake.lua:56]: Yeah the functions hasFatalCompileWarnings and hasFatalLinkerWarnings do the same now. I was in two minds about collapsing both into hasFatalWarnings and use them in both places. Is that what you meant?

File: website/docs/linkerfatalwarnings.md (hunk 3 above)
Comment (redorav) [reply website/docs/linkerfatalwarnings.md:23]: Added some simple examples

File: website/docs/linkerfatalwarnings.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+Premake 5.0 or later. Special value `All` available since Premake 5.0-beta5 or later.
+
+### Examples ###
+
+```lua
+filter { "toolset:msc" }
+	fatalwarnings { "4044" }
```
Comment (Jarod42) [inline website/docs/linkerfatalwarnings.md:31]:
```suggestion
filter { "toolset:msc" }
	fatalwarnings { "4044" } -- unrecognized option 'option'; ignored
filter {}
```

File: website/docs/fatalwarnings.md [hunk 5]
```diff
@@ -10,20 +10,29 @@ fatalwarnings { "warnings" }
...
+Premake 5.0 or later. Special value `All` available since Premake 5.0-beta5 or later.
+
+### Examples ###
+
+```lua
+filter { "toolset:msc" }
+	fatalwarnings { "4035" }
+
+filter { "toolset:clang" }
+	fatalwarnings { "-Wreturn-type" }
```
Comment (Jarod42) [inline website/docs/fatalwarnings.md:34]:
```suggestion
filter { "toolset:msc" }
	fatalwarnings { "4035" } -- 'function': no return value

filter { "toolset:clang" }
	fatalwarnings { "-Wreturn-type" }

filter {}
```
Closing scope, someone even would have preferred extra `filter {}` before opening the other one ;-)
I also add comment as number is not self explanatory,
even if here, with clang part, we might expect same warning.

File: website/docs/linkerfatalwarnings.md (hunk 4 above)
Comment (redorav) [reply website/docs/linkerfatalwarnings.md:31]: Those are good suggestions, I don't love the closing filter as it's a bit of noise but don't mind it too much. Not on laptop atm but will commit as soon as I can

File: website/docs/fatalwarnings.md (hunk 5 above)
Comment (redorav) [reply website/docs/fatalwarnings.md:34]: Also good suggestions, same comments as the other one apply :)

File: tests/test_premake.lua (hunk 2 above)
Comment (redorav) [reply tests/test_premake.lua:56]: Let me know what it is you want here, I'll resolve for now

**Comment by Jarod42** [comment]
Failed
>  Config file: solaris-11.4-gcc.conf
>  VM is booting
>  VM is booting

but unrelated to the PR.
30 minutes timeout seems long for something which seems to fail forever or success quickly.

**Review by Jarod42 — APPROVED** [review]

**Review by nickclark2016 — APPROVED** [review]

**Pull request merged**